File: kaltura/api_v3/generator/dotnet/output/KalturaClient/Types/KalturaUser.cs
using System;
using System.Xml;
using System.Collections.Generic;

namespace Kaltura
{
	public class KalturaUser : KalturaObjectBase
	{
		#region Properties
		public string Id = null;
		public string ScreenName = null;
		public string FullName = null;
		public string Email = null;
		public string DateOfBirth = null;
		public string Country = null;
		public string State = null;
		public string City = null;
		public string Zip = null;
		public string UrlList = null;
		public string Picture = null;
		public int Icon = Int32.MinValue;
		public string AboutMe = null;
		public string Tags = null;
		public string MobileNum = null;
		public int Gender = Int32.MinValue;
		public int Views = Int32.MinValue;
		public int Fans = Int32.MinValue;
		public int Entries = Int32.MinValue;
		public int ProducedKshows = Int32.MinValue;
		public int Status = Int32.MinValue;
		public int CreatedAt = Int32.MinValue;
		public int UpdatedAt = Int32.MinValue;
		public int PartnerId = Int32.MinValue;
		public int DisplayInSearch = Int32.MinValue;
		public int PartnerData = Int32.MinValue;
		#endregion

		#region CTor
		public KalturaUser()
		{
		}

		public KalturaUser(XmlElement node)
		{
			foreach (XmlElement propertyNode in node.ChildNodes)
			{
				string txt = propertyNode.InnerText;
				switch (propertyNode.Name)
				{
					case "id":
						this.Id = txt;
						continue;
					case "screenName":
						this.ScreenName = txt;
						continue;
					case "fullName":
						this.FullName = txt;
						continue;
					case "email":
						this.Email = txt;
						continue;
					case "dateOfBirth":
						this.DateOfBirth = txt;
						continue;
					case "country":
						this.Country = txt;
						continue;
					case "state":
						this.State = txt;
						continue;
					case "city":
						this.City = txt;
						continue;
					case "zip":
						this.Zip = txt;
						continue;
					case "urlList":
						this.UrlList = txt;
						continue;
					case "picture":
						this.Picture = txt;
						continue;
					case "icon":
						this.Icon = ParseInt(txt);
						continue;
					case "aboutMe":
						this.AboutMe = txt;
						continue;
					case "tags":
						this.Tags = txt;
						continue;
					case "mobileNum":
						this.MobileNum = txt;
						continue;
					case "gender":
						this.Gender = ParseInt(txt);
						continue;
					case "views":
						this.Views = ParseInt(txt);
						continue;
					case "fans":
						this.Fans = ParseInt(txt);
						continue;
					case "entries":
						this.Entries = ParseInt(txt);
						continue;
					case "producedKshows":
						this.ProducedKshows = ParseInt(txt);
						continue;
					case "status":
						this.Status = ParseInt(txt);
						continue;
					case "createdAt":
						this.CreatedAt = ParseInt(txt);
						continue;
					case "updatedAt":
						this.UpdatedAt = ParseInt(txt);
						continue;
					case "partnerId":
						this.PartnerId = ParseInt(txt);
						continue;
					case "displayInSearch":
						this.DisplayInSearch = ParseInt(txt);
						continue;
					case "partnerData":
						this.PartnerData = ParseInt(txt);
						continue;
				}
			}
		}
		#endregion

		#region Methods
		public new KalturaParams ToParams()
		{
			KalturaParams kparams = base.ToParams();
			kparams.AddStringIfNotNull("id", this.Id);
			kparams.AddStringIfNotNull("screenName", this.ScreenName);
			kparams.AddStringIfNotNull("fullName", this.FullName);
			kparams.AddStringIfNotNull("email", this.Email);
			kparams.AddStringIfNotNull("dateOfBirth", this.DateOfBirth);
			kparams.AddStringIfNotNull("country", this.Country);
			kparams.AddStringIfNotNull("state", this.State);
			kparams.AddStringIfNotNull("city", this.City);
			kparams.AddStringIfNotNull("zip", this.Zip);
			kparams.AddStringIfNotNull("urlList", this.UrlList);
			kparams.AddStringIfNotNull("picture", this.Picture);
			kparams.AddIntIfNotNull("icon", this.Icon);
			kparams.AddStringIfNotNull("aboutMe", this.AboutMe);
			kparams.AddStringIfNotNull("tags", this.Tags);
			kparams.AddStringIfNotNull("mobileNum", this.MobileNum);
			kparams.AddIntIfNotNull("gender", this.Gender);
			kparams.AddIntIfNotNull("views", this.Views);
			kparams.AddIntIfNotNull("fans", this.Fans);
			kparams.AddIntIfNotNull("entries", this.Entries);
			kparams.AddIntIfNotNull("producedKshows", this.ProducedKshows);
			kparams.AddIntIfNotNull("status", this.Status);
			kparams.AddIntIfNotNull("createdAt", this.CreatedAt);
			kparams.AddIntIfNotNull("updatedAt", this.UpdatedAt);
			kparams.AddIntIfNotNull("partnerId", this.PartnerId);
			kparams.AddIntIfNotNull("displayInSearch", this.DisplayInSearch);
			kparams.AddIntIfNotNull("partnerData", this.PartnerData);
			return kparams;
		}
		#endregion
	}
}

File: kaltura/api_v3/lib/types/KalturaUser.php
<?php
/**
 * @package api
 * @subpackage objects
 */
class KalturaUser extends KalturaObject 
{
	/**
	 * @var string
	 * @readonly
	 * @filter eq,not,in,notIn,gte
	 */
	public $id;

	/**
	 * @var string
	 */
	public $screenName;

	/**
	 * @var string
	 */
	public $fullName;

	/**
	 * @var string
	 */
	public $email;

	/**
	 * @var string
	 */
	public $dateOfBirth;
	
	/**
	 * @var string
	 */
	public $country;

	/**
	 * @var string
	 */
	public $state;

	/**
	 * @var string
	 */
	public $city;

	/**
	 * @var string
	 */
	public $zip;
	
	/**
	 * @var string
	 */
	public $urlList;
	
	/**
	 * @var string
	 */
	public $picture;
	
	/**
	 * @var int
	 * @readonly
	 */
	public $icon;
	
	/**
	 * @var string
	 */
	public $aboutMe;
	
	/**
	 * @var string
	 */
	public $tags;
	
	/**
	 * @var string
	 */
	public $mobileNum;
	
	/**
	 * @var int
	 */
	private $matureContent;
	
	/**
	 * @var int
	 */
	public $gender;

	/**
	 * @var int
	 * @readonly
	 */
	public $views;

	/**
	 * @var int
	 * @readonly
	 */
	public $fans;

	/**
	 * @var int
	 * @readonly
	 */
	public $entries;

	/**
	 * @var int
	 * @readonly
	 */
	public $producedKshows;

	/**
	 * @var int
	 */
	public $status;

	/**
	 * Entry creation date as Unix timestamp (In seconds)
	 * @var int
	 * @readonly
	 */
	public $createdAt;

	/**
	 * Entry update date as Unix timestamp (In seconds)
	 * @var int
	 * @readonly
	 */
	public $updatedAt;

	/**
	 * @var int
	 * @readonly
	 */
	public $partnerId;

	/**
	 * @var int
	 */
	public $displayInSearch;

	/**
	 * @var int
	 */
	public $partnerData;

	private static $map_between_objects = array
	(
		"id" => "puserId" , "screenName" , "fullName" , "email" , "dateOfBirth" , "country" , "state" , "city" ,
		"zip" , "urlList" , "picture" , "icon" , "aboutMe" , "tags" , "mobileNum" , "gender" , "views" ,
		"fans" , "entries" , "producedKshows" , "status" , "createdAt" , "updatedAt" , "partnerId" , "displayInSearch" ,  "partnerData" 
	);

	public function getMapBetweenObjects ( )
	{
		return array_merge ( parent::getMapBetweenObjects() , self::$map_between_objects );
	}

	public function fromUser ( kuser $entry )
	{
		parent::fromObject( $entry );
	}
	
	public function toUser () 
	{
		$user = new kuser();
		return parent::toObject( $user );
	}
	
	public static function getKuserByPuserId( $puser_id, $partner_id )
	{
		$puserKuser = PuserKuserPeer::retrieveByPartnerAndUid( $partner_id , null, $puser_id );
		if ( $puserKuser )
		{
			$userFromDb = kuserPeer::retrieveByPK( $puserKuser->getKuserId() );
		}
		return $userFromDb;
	}	
}
?>